source/common/common/utility.cc

```diff
@@ -134,7 +134,7 @@ bool StringUtil::caseCompare(absl::string_view lhs, absl::string_view rhs) {
   if (rhs.size() != lhs.size()) {
     return false;
   }
-  return absl::strings_internal::memcasecmp(rhs.data(), lhs.data(), rhs.size()) == 0;
+  return absl::StartsWithIgnoreCase(rhs.data(), lhs.data());
```

Comment:
try swapping lhs and rhs.

Reply:
@jmarantz  gzip_filter_test passed after removing ` .data() ` calls
https://gist.github.com/athampy/545550bbd2b1bcb461cb9d677ef802f2

Reply:
Right! @athampy absl::StartsWithIgnoreCase expects absl::string_view while var.data() returns a char *

Reply:
Yes, exactly!

Here's what I think is happening: 

`absl::string_view.size()` doesn't necessarily represent the size of the underlying `char*`
When we pass in `rhs.data()`, for example, it gets implicitly converted to an `absl::string_view` where the size() == size of the char*

This becomes a problem when we have:

```bash
lhs = <absl::string_view> (_ptr="my;test;string", size=3)
// lhs.data() is "my;test;string" which gets converted to absl::string_view with size=14

rhs = <absl::string_view> (_ptr="my;", size=3)
// rhs.data() is "my;" which gets converted to absl::string_view with size=3

absl::StartsWithIgnoreCase(rhs.data(), lhs.data()) -> false // rhs.size() is not >= lhs.size()
```